From 7a553e6aa66f11349172b66432cf451520dbeb00 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <panto@antoniou-consulting.com>
Date: Wed, 2 Sep 2015 16:36:38 -0400
Subject: [PATCH 547/800] HACK: ARM: Export cache flush management symbols
 when !MULTI_CACHE

Imagination DDK 1.9 uses dma APIs that are supposed to be private to
dmaengine and not used directly as explained in
https://lkml.org/lkml/2013/1/3/432.

However, this is fixed only in DDK1.12 or later.

For the moment, export the functions as a hack. This patch will be
reverted upon migration to DDK1.14.

Signed-off-by: Pantelis Antoniou <panto@antoniou-consulting.com>
Signed-off-by: Tomi Valkeinen <tomi.valkeinen@ti.com>
Signed-off-by: Eric Ruei <e-ruei1@ti.com>
Signed-off-by: Jyri Sarha <jsarha@ti.com>
[zou: Original patch taken from PROCESSOR-SDK-LINUX-AM335X 02_00_01_07]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 arch/arm/kernel/setup.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/arch/arm/kernel/setup.c b/arch/arm/kernel/setup.c
index 14e5dc2..3befb18 100644
--- a/arch/arm/kernel/setup.c
+++ b/arch/arm/kernel/setup.c
@@ -1138,3 +1138,12 @@ const struct seq_operations cpuinfo_op = {
 	.stop	= c_stop,
 	.show	= c_show
 };
+
+/* export the cache management functions */
+#ifndef MULTI_CACHE
+
+EXPORT_SYMBOL(__glue(_CACHE, _dma_map_area));
+EXPORT_SYMBOL(__glue(_CACHE, _dma_unmap_area));
+EXPORT_SYMBOL(__glue(_CACHE, _dma_flush_range));
+
+#endif
-- 
1.7.5.4

